napari/layers/tracks/tracks.py

```diff
@@ -418,59 +419,61 @@ def properties(self) -> Dict[str, np.ndarray]:
         """dict {str: np.ndarray (N,)}: Properties for each track."""
         return self._manager.properties
 
-    @property
-    def properties_to_color_by(self) -> List[str]:
-        """track properties that can be used for coloring etc..."""
-        return list(self.properties.keys())
-
     @properties.setter
     def properties(self, properties: Dict[str, np.ndarray]):
         """set track properties"""
         self.features = properties
 
     @property
-    def graph(self) -> Dict[int, Union[int, List[int]]]:
+    def properties_to_color_by(self) -> List[str]:
+        """track properties that can be used for coloring etc..."""
+        return list(self.properties.keys())
+
+    @property
+    def graph(self) -> Optional[Dict[int, List[int]]]:
         """dict {int: list}: Graph representing associations between tracks."""
         return self._manager.graph
 
     @graph.setter
     def graph(self, graph: Dict[int, Union[int, List[int]]]):
         """Set the track graph."""
-        self._manager.graph = graph
+        # Ignored type, because mypy can't handle different signatures
+        # on getters and setters; see https://github.com/python/mypy/issues/3004
+        self._manager.graph = graph  # type: ignore[assignment]
         self._manager.build_graph()
         self.events.rebuild_graph()
 
     @property
-    def tail_width(self) -> Union[int, float]:
+    def tail_width(self) -> int:
         """float: Width for all vectors in pixels."""
         return self._tail_width
 
     @tail_width.setter
-    def tail_width(self, tail_width: Union[int, float]):
-        self._tail_width = np.clip(tail_width, 0.5, self._max_width)
+    def tail_width(self, tail_width: int):
+        self._tail_width: int = np.clip(tail_width, 0.5, self._max_width)
```

Comment:
This call to `np.clip` will return a numpy scalar (e.g. `np.int64` or `np.float64`). Given that minimum tail width is 0.5, I think declaring/annotating `tail_width` as a `float` rather than an `int` seems like the way to go. Then cast that to a `float`, if needed.

In general, I think most of the numerical quantities changed in this PR that were previously typed as `Union[int, float]` should probably just be `float`. The default values are integers, but I don't see why the attributes/properties can't be floating point.